### ISSUE TITLE
utf-8-bom is not supported

### ISSUE THREAD

**Issue by kazuki-ma** [ec4j/editorconfig-maven-plugin#35]
Try with following `.editorconfig`

```

root = true

[*.{cs,shader}]
charset = utf-8-bom
end_of_line = lf
indent_size = 4
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true
```

Leads following error.

```
11:24:16.809 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.809 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] FAILURE: Build failed with an exception.
11:24:16.809 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * What went wrong:
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Execution failed for task ':editorconfigFormat'.
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] > There was a failure while executing work items
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]    > A failure occurred while executing org.ec4j.gradle.runtime.EditorconfigInvoker
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]       > utf-8-bom
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Try:
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]  Run with --scan to get full insights.
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.810 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Exception is:
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':editorconfigFormat'.
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:96)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:65)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ActionEventFiringTaskExecuter.execute(ActionEventFiringTaskExecuter.java:44)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.TimeoutTaskExecuter.execute(TimeoutTaskExecuter.java:53)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SnapshotAfterExecutionTaskExecuter.execute(SnapshotAfterExecutionTaskExecuter.java:38)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:49)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:61)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
11:24:16.811 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:44)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:62)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:55)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:35)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.run(EventFiringTaskExecuter.java:49)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:44)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:43)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:337)
11:24:16.812 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:325)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:318)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:304)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:134)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker$1.execute(DefaultPlanExecutor.java:129)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:202)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:193)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:129)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: org.gradle.workers.WorkerExecutionException: There was a failure while executing work items
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor.workerExecutionException(DefaultWorkerExecutor.java:161)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor.await(DefaultWorkerExecutor.java:155)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.ec4j.gradle.AbstractEditorconfigTask.perform(AbstractEditorconfigTask.java:128)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:48)
11:24:16.813 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:41)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:704)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:671)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:117)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:301)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:293)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:106)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:85)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 35 more
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: org.gradle.workers.internal.DefaultWorkerExecutor$WorkExecutionException: A failure occurred while executing org.ec4j.gradle.runtime.EditorconfigInvoker
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor$WorkerExecution.waitForCompletion(DefaultWorkerExecutor.java:287)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForItemsAndGatherFailures(DefaultAsyncWorkTracker.java:115)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.work.DefaultAsyncWorkTracker.waitForCompletion(DefaultAsyncWorkTracker.java:87)
11:24:16.814 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor.await(DefaultWorkerExecutor.java:153)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 50 more
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] Caused by: java.nio.charset.UnsupportedCharsetException: utf-8-bom
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.ec4j.gradle.runtime.EditorconfigInvoker.run(EditorconfigInvoker.java:145)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:39)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$WorkerCallable.call(IsolatedClassloaderWorkerFactory.java:177)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory.executeInWorkerClassLoader(IsolatedClassloaderWorkerFactory.java:105)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory.access$100(IsolatedClassloaderWorkerFactory.java:55)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1$1.call(IsolatedClassloaderWorkerFactory.java:79)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1$1.call(IsolatedClassloaderWorkerFactory.java:76)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:315)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:305)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:175)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:101)
11:24:16.815 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.IsolatedClassloaderWorkerFactory$1.execute(IsolatedClassloaderWorkerFactory.java:76)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor$1.call(DefaultWorkerExecutor.java:108)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.workers.internal.DefaultWorkerExecutor$1.call(DefaultWorkerExecutor.java:102)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]   ... 3 more
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter]
11:24:16.816 [ERROR] [org.gradle.internal.buildevents.BuildExceptionReporter] * Get more help at https://help.gradle.org
11:24:16.817 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger]
11:24:16.817 [ERROR] [org.gradle.internal.buildevents.BuildResultLogger] BUILD FAILED in 0s
```

**Comment by ppalaga** [comment]
Thanks for the report, @kazuki-ma. Closing here and creating https://github.com/ec4j/ec4j/issues/93 in ec4j project where the fix would eventually have to happen.

**Comment by kazuki-ma** [comment]
@ppalaga Got it. Thank you for you good handling!
I'll watch ec4j issue.

**Comment by ppalaga** [comment]
In the end, I managed to provide a fix. I just released editorconfig-maven-plugin 0.0.10 - it will take tens of minutes till it appears on Maven Central. Please check that you see 0.0.10 there before using it: http://central.maven.org/maven2/org/ec4j/maven/editorconfig-maven-plugin/

Please report any issues if the fix does not work for you.
